lib/SILOptimizer/Utils/CanonicalizeOSSALifetime.cpp

```diff
@@ -249,6 +250,33 @@ void CanonicalizeOSSALifetime::extendLivenessToDeinitBarriers() {
   SmallVector<SILInstruction *, 4> outsideDestroys;
   findDestroysOutsideBoundary(outsideDestroys);
 
+  SmallVector<SILBasicBlock *, 32> discoveredBlocks(this->discoveredBlocks);
+  SSAPrunedLiveness completeLiveness(*liveness, &discoveredBlocks);
+
+  for (auto *end : outsideDestroys) {
+    completeLiveness.updateForUse(end, /*lifetimeEnding*/ true);
+  }
+
+  OSSALifetimeCompletion::visitUnreachableLifetimeEnds(
+      getCurrentDef(), completeLiveness, [&](auto *unreachable) {
+        recordConsumingUser(unreachable);
+        if (auto *previous = unreachable->getPreviousInstruction()) {
+          if (liveness->isInterestingUser(previous) ==
```

Comment:
Let's avoid this bug-prone check in a separate refactoring. Maybe PrunedLiveness needs an extendToNonUse API

Reply:
https://github.com/apple/swift/pull/68756